runtime/lua/vim/lsp.lua

```diff
@@ -1529,6 +1529,48 @@ function lsp.omnifunc(findstart, base)
   return -2
 end
 
+--- Provides an interface between the built-in client and a `formatexpr` function.
+--- Currently only supports a single client.
+--@param start_line 1-indexed line, defaults to |v:lnum|
+--@param end_line 1-indexed line, calculated based on {start_line} and |v:count|
+--@param timeout_ms optional, defaults to 500ms
```

Comment:
Should we point out in the docs how one would configure this? It can't be set directly using  `setlocal formatexpr=v:lua..`, right?

Reply:
That actually does work

Reply:
Sorry I should have been more specific - I was referring to the `timeout_ms` value

Reply:
Oh, yes. Also made it an opt, added a small note in the docs. wdyt?